scripts/rematch_reads_to_haplotypes.py

```diff
@@ -0,0 +1,318 @@
+    best_hap = None
+    best_start_point = None
+    best_end_point = None
+    read_seq = read.query_sequence
+    best_hap_start_position = None
+    best_hap_seq = None
+
+    read_start_position = read.reference_start - sc_length_start
+    read_end_position = read.reference_end + sc_length_end
+    # extract the region of the assembly that the read overlaps
+    for hap in region_haps:
+        # only in case we overlap the breakpoint
+        # read.reference_start is the breakpoint position
+        hap_sc_size_start = hap.cigar[0][1] if hap.cigar[0][0] == 4 else 0
+        hap_sc_size_end = hap.cigar[-1][1] if hap.cigar[-1][0] == 4 else 0
+
+        hap_seq = hap.query_sequence
+        hap_start_position = hap.reference_start - hap_sc_size_start
+        hap_end_position = hap.reference_end + hap_sc_size_end
+
+        # local alignment
+        if((read_start_position <= hap_end_position) and
+            (read_end_position >= hap_start_position) and
+            ((sc_length_start == 0 and sc_length_end == 0) or (sc_length_start > 0 and read.reference_start >= hap_start_position) or
+            (sc_length_end > 0 and read.reference_end <= hap_end_position))):
+
+            score, _, _ = run_alignment(hap_seq,
+                                          read_seq,
+                                          hap_start_position,
+                                          0,
+                                          hap.cigartuples,
+                                          True)
+            if score > best_score:
+                best_score = score
+                best_hap = hap
+                best_hap_seq = hap_seq
+                best_hap_start_position = hap_start_position
+
+                affected_haps.add((hap.query_name, hap.flag))
+
+    # check reference sequence as well
+    del_length = sum([length for op, length in read.cigartuples if op == 2])
+
+    ref_seq = reference[read.reference_name][
+             max(read.reference_start - sc_length_start, 0):
+             min(read.reference_end + sc_length_end + del_length, len(reference[read.reference_name]))].seq.upper()
+    ref_score, _, _ = run_alignment(ref_seq,
+                                    read_seq,
+                                    read_start_position,
+                                    0,
+                                    [],
+                                      True)
+    if ref_score > best_score:
+        best_score = ref_score
+        best_hap = None
+    else:
+        # Rerun best haplotype alignment with the best score
+        score, start_pos_local, end_pos_local = run_alignment(best_hap_seq,
+                                      read_seq,
+                                      best_hap_start_position,
+                                      0,
+                                      best_hap.cigartuples,
+                                       False)
+
+        best_start_point = start_pos_local - best_hap_start_position
```

Comment:
I don't understand why positions on the reference are involved here. Looks like a possible bug